### ISSUE TITLE
Bugs in filters_test.py

### ISSUE THREAD

**Issue by WindQAQ** [tensorflow/addons#199]
In `image/filters_test.py`, there are some buggy test cases:

1. There are some patterns like
https://github.com/tensorflow/addons/blob/6bb0d4bfb814df9d9e6d5b200128f25120e09e21/tensorflow_addons/image/filters_test.py#L60-L63
However, this is not safe because each context manager will catch only one exception, and then exit. That it, the following code snippet will pass the test:
```python
def foo(x):
    if x < 5:
        raise ValueError("fail " + str(x))


class Test(tf.test.TestCase):

    def test_context_manager(self):
        with self.assertRaisesRegexp(ValueError, "fail 3"):
            foo(3)
            foo(2)
            foo(1)

        with self.assertRaises(ValueError):
            foo(5)
            foo(4)
            foo(3)
```
2. One test case disables eager execution, but this will affect **all test cases** (in the same file) executed after it.
https://github.com/tensorflow/addons/blob/6bb0d4bfb814df9d9e6d5b200128f25120e09e21/tensorflow_addons/image/filters_test.py#L71-L75
In other words, `test_check_eager` will fail in this toy test case.
```python
class Test(tf.test.TestCase):

    # Make it execute first
    def test_a(self):
        tf.compat.v1.disable_eager_execution()
        tf.compat.v1.placeholder(tf.float32, shape=(1, 2, None))

    def test_check_eager(self):
        if not tf.executing_eagerly():
            raise Exception("Not in eager mode")
```
This might not be what we want. In order to utilize `tf.compat.v1.placeholder`, which can be used to test `None` shape, in tf2.0, maybe `@run_deprecated_v1` is helpful enough.

**Comment by facaiy** [comment]
Tzu-Wei, would you like to fix them?
